recipes/cuda-cccl-impl/meta.yaml

```
@@ -0,0 +1,67 @@
+{% set name = "cuda-cccl-impl" %}
+{% set version = "2.0.1" %}
+{% set thrust_version = "2.0.1" %}
+{% set cub_version = "2.0.1" %}
+{% set libcudacxx_version = "1.9.0" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.gz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/NVIDIA/thrust/archive/refs/tags/{{ thrust_version }}.{{ extension }}
+    folder: thrust
+    sha256: 06df373cf487cf3710be6798bc5441a03685b78e5c1338e2f14005f7ccd030e6  # [linux]
+    sha256: 39a9fc88b0b06da118cc3ed28cbfca27b2a7ae192877996cab1295176ceed65e  # [win]
+  - url: https://github.com/NVIDIA/cub/archive/refs/tags/{{ cub_version }}.{{ extension }}
+    folder: cub
+    sha256: beeef0962c87eca994db06cb9e9168a71c63fb369398a5b5f1db923140814701  # [linux]
+    sha256: 22308facf62421d6dd2489b9554ffa4908dca13d3921a3e7f2965c5a6984cd20  # [win]
+  - url: https://github.com/NVIDIA/libcudacxx/archive/refs/tags/{{ libcudacxx_version }}.{{ extension }}
+    folder: libcudacxx
+    sha256: 05b1435ad65f5bdef1bb8d1eb29dc8f0f7df34c01d77bf8686687a4603588bce  # [linux]
+    sha256: d7864460ce8def8e0620cb9dd891e94299ade75eac75aa96b34001af3cdb074c  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+
```

@isuruf @kkraus14 @adibbley I think this is the constraint we are needing.
```suggestion

requirements:
  run_constrained:
    - cub < 0.0.0
    - thrust < 0.0.0

```

---

It would probably be cleaner to have a run constraint and repodata patch on the deprecated `thrust` and `cub` packages so that this `cuda-cccl` package doesn't need to have this `cub` and `thrust` run constraint in perpetuity, but whatever is decided upon I'm happy either way.

---

Agreed, @kkraus14. I would propose adding the constraint on this recipe for now, to have an immediate solution. I agree that the patch would be better to apply on the deprecated packages. However, I don't have the knowledge (or merge-power) to help with implementing that option.

---

I think we could avoid needing to modify the Thrust / cub recipes where once we merge this we can archive them.

I think we'd just need a pull request to the repodata patches repository: https://github.com/conda-forge/conda-forge-repodata-patches-feedstock which I imagine would look something like:
```
if record_name in {"thrust", "cub"}:
    new_constrains = record.get('constrains', [])
    new_constrains.append("cuda-cccl < 0.0.0a0")
    new_constrains.append("cuda-cccl-impl < 0.0.0a0")
    new_constrains.append(f"cuda-cccl_{subdir} < 0.0.0a0")
    record['constrains'] = new_constrains
```

I have merge-power and am happy to help review + iterate through merging.